src/brsTypes/Callable.ts

```diff
@@ -217,91 +222,82 @@ export class Callable implements Brs.BrsValue {
         return this.name || "";
     }
 
-    getFirstSatisfiedSignature(args: Brs.BrsType[]): SignatureAndImplementation | undefined {
-        return this.signatures.filter(
-            (sigAndImpl) => this.getSignatureMismatches(sigAndImpl.signature, args).length === 0
-        )[0];
+    /**
+     * Attempts to satisfy each signature of this Callable using the provided arguments, coercing arguments into other
+     * types where necessary and possible, returning the first satisfied signature and the arguments.
+     * @param args - the arguments to satisfy this Callable with
+     * @returns the signature, implementation, type mismatches, and coerced arguments for the first encountered
+     *          signature satisfied by the provide arguments.
+     */
+    getFirstSatisfiedSignature(args: Brs.BrsType[]): SignatureSatisfaction | undefined {
+        for (let sigAndImpl of this.signatures) {
+            let satisfaction = this.trySatisfySignature(sigAndImpl, args);
+            if (satisfaction.mismatches.length === 0) {
+                return satisfaction;
+            }
+        }
+        return undefined;
     }
 
-    getAllSignatureMismatches(args: Brs.BrsType[]): SignatureAndMismatches[] {
-        return this.signatures.map((sigAndImpl) => ({
-            signature: sigAndImpl.signature,
-            mismatches: this.getSignatureMismatches(sigAndImpl.signature, args),
-        }));
+    /**
+     * Attempts to satisfy each signature of this Callable using the provided arguments, coercing arguments into other
+     * types where necessary and possible.
+     * @param args - the arguments to satisfy this Callable with
+     * @returns the signature, implementation, type mismatches, and coerced arguments for each signature in this
+     *          Callable.
+     */
+    getAllSignatureMismatches(args: Brs.BrsType[]): SignatureSatisfaction[] {
+        return this.signatures.map((sigAndImpl) => this.trySatisfySignature(sigAndImpl, args));
     }
 
-    private getSignatureMismatches(sig: Signature, args: Brs.BrsType[]): SignatureMismatch[] {
+    private trySatisfySignature(
```

Comment:
The name of the function is pretty self explanatory but could you add docs please?